variables.tf

```diff
@@ -166,25 +166,14 @@ variable "restrict_public_buckets" {
   description = "Set to `false` to disable the restricting of making the bucket public"
 }
 
-variable "s3_replication_enabled" {
-  type        = bool
-  default     = false
-  description = "Set this to true and specify `s3_replica_bucket_arn` to enable replication. `versioning_enabled` must also be `true`."
-}
-
-variable "s3_replica_bucket_arn" {
-  type        = string
-  default     = ""
-  description = "The ARN of the S3 replica bucket (destination)"
-}
-
```

Comment:
I will need to weigh in on @Nuru and @osterman to decide on whether removing these outright is the best upgrade path for our module's users.

In this particular case I feel like `var.s3_replication_enabled` could be deprecated instead of outright removed, whereas `var.s3_replica_bucket_arn` cannot be. But in that case, maybe it doesn't make sense to deprecate and keep one without the other.

Reply:
Let me know.

As someone who went through the change of lifecycle rules from individual variables to the `lifecycle_rules` combined structure, my experience was that the transition was significantly harder because the old variables still existed but did nothing. I think it's easier for users to simply break compatibility and describe the migration path in release notes.